TESTS/network/wifi/wifi_tests.h

```diff
@@ -0,0 +1,26 @@
+#ifndef WIFI_TESTS_H
+#define WIFI_TESTS_H
+
+#include "WiFiInterface.h"
+
+WiFiInterface *get_interface(void);
+
+/* Test cases */
+void wifi_constructor(void);
```

Comment:
To be aware, we are enabling doxygen generation for tests (https://github.com/ARMmbed/mbed-os/pull/5238). It would be beneficial to have these test cases documented here